### PR TITLE
Rework OCI packaging

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR reworks the way we package our remote updater payload.
We now always build both AMD and ARM versions of the agent, and package those to a single OCI artifact using the tool provided by the folks from the Remote Updater / Fleet automation team.

### Motivation

Providing payloads for the remote updater

### Additional Notes

Related omnibus-ruby PR: https://github.com/DataDog/omnibus-ruby/pull/191

### Possible Drawbacks / Trade-offs

This adds an arm build to the default configuration. I couldn't think of a way to conditionally enable "needs" for the packaging job, but my hope is that we soon split the packaging stage outside of the build one, which would greatly reduce the number of builds, and would make this an innocuous change.

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

**Review by bkabrda — APPROVED** [review]

**Review by arbll — APPROVED** [review]

**Review by richl-dd — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

File: .gitlab/packaging/include.yml
[the remark this answers is not in the document]
Comment (chouquette) [reply .gitlab/packaging/include.yml:1]: Good point, done

**Pull request merged**